Source/WebCore/dom/Document.cpp

```diff
@@ -383,6 +384,33 @@ struct Document::PendingScrollEventTargetList {
 
 static const Seconds intersectionObserversInitialUpdateDelay { 2000_ms };
 
+class CallbackForContainIntrinsicSize : public NativeResizeObserverCallback {
+public:
+    void onResize(const Vector<Ref<ResizeObserverEntry>>&) final;
+};
+
+void CallbackForContainIntrinsicSize::onResize(const Vector<Ref<ResizeObserverEntry>>& entries)
+{
+    for (auto& entry : entries) {
+        if (auto* target = entry->target()) {
+            if (!target->isConnected()) {
+                target->document().unobserveForContainIntrinsicSize(*target);
+                continue;
+            }
+
+            if (auto* box = target->renderBox(); !box || box->shouldApplySizeContainment()) {
+                auto* lastRememberedSize = target->lastRememberedSize();
+                LayoutSize size = lastRememberedSize ? LayoutSize(lastRememberedSize->inlineSize(), lastRememberedSize->blockSize()) : LayoutSize(-1, -1);
+                target->document().updateObservationSizeForContainIntrinsicSize(*target, size);
```

Comment:
I don't understand the logic here. If the element doesn't have a last remembered size, then this sets the observation to -1x-1 to schedule another notification, which will likely trigger the same logic and keep running this code every frame even if nothing changes? Why?
And  if the element has a last remembered size from before getting size containment, this sets the observation size to the last remembered size, which is probably different, so it also schedules notifications.

Reply:
Yes, that's true, it will be triggered every frame if there is no change.
From this point of view, look like the "unobserve" solution is move efficient.
OK, I will work on it:)